### DISCUSSION TITLE
Registering custom dataset.

### DISCUSSION THREAD

**Discussion by Subhajyotidas** [open-mmlab/mmocr#1815] [General]
I started preparing for text recognition training with japanese dataset for mmocr.
I created the synthetic japanese dataset and annotation according to the example of toy dataset.
One problem I am stuck with is how to register the custom dataset for mmocr.
I did not find any concrete example of registering dataset with proper prefixes.
Can you please help me out with this problem?
Thank you. 
![image](https://user-images.githubusercontent.com/60061358/228114978-cc3fa6e8-f097-43cc-a1c7-f283561a5c73.png)
The 'Toy' keyword in second last line is the prefix, which has already been registered.
There are other prefixes too, e.g. 'IC15'.

Note: I am trying to train 'sar' model for recognition.

**Comment by gaotongxiao** [comment]
`dataset_prefixes` only affects how the metrics will be logged, which is used as an identifier when multiple datasets are being evaluated. It doesn't matter when you only have one dataset, and you don't have to do anything to "register" the dataset, just use the right class to load your annotations would be fine.

Some resources that might be helpful:
https://mmocr.readthedocs.io/en/dev-1.x/user_guides/config.html#data-related-configuration
https://mmocr.readthedocs.io/en/dev-1.x/user_guides/config.html#evaluator (Read `MultiDatasetsEvaluator`)
https://mmocr.readthedocs.io/en/dev-1.x/user_guides/dataset_prepare.html#dataset-configuration

**Comment by Subhajyotidas** [comment]
Thank you for your reply.  
I am fairly new to the mmocr.
I have structured my dataset in the following manner.

japanese_data
       |
       train
              |
              img1.png
              img2.png
              ...
              label.txt
      test
              |
              img1.png
              img2.png
              ...
              label.txt
The labelling format in same as for Toy_dataset in json format.
Can you please tell me how to define the class for this dataset and load it?
Because I tried to just replace the names of the dataset and config file to train the text recognizer. 
It will be wrong evaluation if we don't replace Toy dataset with proper evaluator dataset. (Tried commenting the evaluator step, gave error.)
Got stuck in the evaluator step...
val_evaluator = dict(dataset_prefixes=['Toy'])
test_evaluator = val_evaluator
To be precise, how to load the new evaluator without prefix?

**Reply by gaotongxiao** [reply]
What's the error message?

**Reply by Subhajyotidas** [reply]
![image](https://user-images.githubusercontent.com/60061358/228153531-819d2a37-b301-4d9e-b3e0-7d8c5459a619.png)
This is the error.

**Reply by gaotongxiao** [reply]
`val_evaluator = dict(dataset_prefixes=['Toy'])` should not be commented out. Just assign it a prefix and it should work like a charm.

**Reply by Subhajyotidas** [reply]
Thank you for your reply.
I am just trying to assign prefix for past few days, could not figure it out.

**Reply by gaotongxiao** [reply]
> Because I tried to just replace the names of the dataset and config file to train the text recognizer.

If you are not sure about the consequence of each modification, just replace  `data_root` and `ann_file` from the original config, and leave everything else still. Then it should work without any error.

If time permitted, I still strongly recommend you go through the docs I provided, which could address most of your questions here.

**Reply by Subhajyotidas** [reply]
Thank you.
I have read the docs before, I will again go through it.
As I am tryiing to train for Japanese texts leaving evaluator to an english text dataset will not serve the purpose I think (It will run though but evaluation will not be correct. I have run it).
If you have any info on the registering data or dataset prefixes, please let me know.
